### PR TITLE
Compute entrypoints manifest

### PR DESCRIPTION
**Pull request by shramee**
New `computed` field on Contract in manifest...

```js
      contract.computed = [
        // For `#[computed]`
        {
          "contract": "actions",
          "entrypoint": "entrypoint_name",
          "component": null
        },
        // For `#[computed(ModelName)]`
        {
          "contract": "actions",
          "entrypoint": "entrypoint_name",
          "component": "ModelName"
        }
      ];
```

⚠️ At the moment only FreeFunctions are supported (not the ones in impl).

### CODE REVIEW

File: crates/dojo-lang/src/plugin.rs [hunk 1]
```diff
@@ -65,6 +67,24 @@ impl GeneratedFileAuxData for DojoAuxData {
     }
 }
 
+/// Dojo related auxiliary data of the Dojo plugin.
+#[derive(Debug, Default, PartialEq)]
+pub struct ComputedValuesAuxData {
+    // Name of entrypoint to get computed value
+    pub entrypoint: SmolStr,
+    // Component to compute for
```
Comment (tarrencev) [inline crates/dojo-lang/src/plugin.rs:75]:
```suggestion
    // Model to bind to
```

File: crates/dojo-lang/src/plugin.rs [hunk 2]
```diff
@@ -112,6 +132,77 @@ impl BuiltinDojoPlugin {
...
+        let fn_name = fn_decl.name(db).text(db);
+        let params = fn_decl.signature(db).parameters(db);
+        let param_els = params.elements(db);
+        let mut model = None;
+        if args.len() == 1 {
+            let model_name = args[0].text(db);
+            model = Some(model_name.clone());
+            let model_type =
+                param_els[1].type_clause(db).ty(db).as_syntax_node().get_text(db).replace('@', "");
+            let model_type = model_type.split("::").last().unwrap();
```
Comment (tarrencev) [inline crates/dojo-lang/src/plugin.rs:177]: Hm I feel like there is a safer way to do this with the AST rather than operating directly on the text

**Review by tarrencev — COMMENTED** [review]
Minor feedback but generally looks good

**Review by tarrencev — APPROVED** [review]

File: crates/dojo-lang/src/plugin.rs (hunk 2 above)
Comment (shramee) [reply crates/dojo-lang/src/plugin.rs:177]: Yeah, We could probably get the last path segment.

**Pull request merged**